res/controllers/Denon-Prime-4-scripts.js

```diff
@@ -194,17 +194,49 @@ Prime4.init = function(_id, _debug) {
         type: components.Button.prototype.types.toggle,
     });
 
+    // Sweep FX - Filter Button
+    Prime4.sweepFilter = new components.Button({
+        midi: [0x9F, 0x0C],
+        group: "[QuickEffectRack1_[Channel1]]",
+        key: "enabled",
+        on: 0x02,
+        off: 0x01,
+        input: function(_channel, _control, value, _status, _group) {
+            for (let i = 1; i <= 4; i++) {
+                if (value > 0) {
+                    const effect = "[QuickEffectRack1_[Channel" + i + "]]";
+                    if (engine.getParameter(effect, "enabled") > 0) {
+                        engine.setParameter(effect, "enabled", 0);
+                    } else {
+                        engine.setParameter(effect, "enabled", 1);
+                    }
+                }
+            }
+        },
+    });
```

Comment:
In Mixxx 2.4, you can change the effect being controlled by the SweepFX Knob. You can set that via the `loaded_chain_preset` key IIUC. Have you experimented with that?

Reply:
How do i go about assigning one component to manipulate more than one group? I want the Sweep FX buttons on the console to change the Quick Effect of all four decks, but using `for` loops seems to not behave well with Components JS unless I make four separate components.

Reply:
that's a good point, you can make a custom component that takes care of that, but I don't think that makes a lot of sense since many controllers only have global chain preset settings instead of being per-channel. I think it would make sense to introduce a custom CO for that. I'll look into that later (as well as actually documenting the `loaded_chain_preset` CO).

Reply:
Brilliant, thank you. I'll be sure to implement it once the right building blocks are in place :D

Reply:
Great, feel free to remind me in case I forget I filed #11371 to help me keep track of it too.